test/integration/application_parser_test.rb

```
@@ -0,0 +1,315 @@
+			assert_equal ApplicationVariables::PERSON_INPUTS.select { |i| i[:group] != :applicant && i[:group] != :person && i[:group] != :relationship }.count, 0 
+
+			# nuking a required variable should raise an error
+			assert_raises RuntimeError do 
+				@json_application['People'][0]['Applicant Age'] = nil 
+				read_json!
+			end
+			@json_application['People'][0]['Applicant Age'] = 40
+			read_json! 
+
+			# if you set a conflicting datatype, it should raise an error 
+			# ERR: Doesn't do this, at least for applicant age? But seems to for required_if applicant inputs
+			# assert_raises RuntimeError do 
+				# @json_application['People'][0]['Applicant Age'] = 'Yolo'
+				# read_json!
+			# end
+			# @json_application['People'][0]['Applicant Age'] = 40
+			# read_json! 
+
+			teardown_app app
+	 	end
+
+	 	test "get and process relationships properly #{app[:name]}" do 
+	 		setup_app app 
+
+	 		@people.each do |person|
+	 			json_person = @json_application['People'].find { |a| a['Person ID'] == person.person_id }
+	 			
+	 			assert_equal person.relationships.count, @people.count - 1
+
+ 				json_person['Relationships'].each do |j_p|
+ 					# every person in the json blob should show up in person.relationships
+ 					refute_nil person.relationships.find { |rel| rel.person.person_id == j_p['Other ID'] } 
+ 				end
+
+				# TODO: Test relationship code setting from ApplicationVariables
+
+ 				person.relationships.each do |rel|
+ 					other_person = @people.find { |p| p.person_id == rel.person.person_id }.relationships.find { |p| p.person.person_id == person.person_id }
+ 					# super clunky way to test relationship inverses
+ 					assert_equal ApplicationVariables::RELATIONSHIP_INVERSE[rel.relationship_type], other_person.relationship_type
+ 					refute_nil rel.relationship_attributes # every one should have 'attest primary responsibility'
+ 				end
+	 		end
+
+	 		# ERR: this should raise an error, right? 
```

Yep, should raise an error